src/GHCMod.hs

```diff
@@ -333,7 +342,14 @@ main = handler $ do
 
 progMain :: (Options,[String]) -> IO ()
 progMain (globalOptions,cmdArgs) = do
-    (res,_) <- runGhcModT globalOptions $ ghcCommands cmdArgs
+    let
```

Comment:
This *is* a hack (kind of). Problem is, it's not possible to read from stdin while parsing options (which is needed for memory-mapped files only in context of ghc-mod standalone), so I do it here. Library calls etc don't read from stdin, and interactive code doesn't rely on parsed CLI options. So there.